File: frontend/src/types/reserva-insumo.ts
/** TypeScript interfaces for ReservaInsumo and ConsumoInsumoMissao matching Pydantic schemas. */

export interface ReservaInsumoResponse {
  id: number;
  missao_id: number;
  insumo_id: number;
  quantidade_prevista: number;
  unidade_medida: string;
  justificativa_excesso: string | null;
  created_at: string;
}

export interface ReservaInsumoCreate {
  missao_id: number;
  insumo_id: number;
  quantidade_prevista: number;
  unidade_medida: string;
  justificativa_excesso?: string | null;
}

export interface ConsumoInsumoMissaoResponse {
  id: number;
  missao_id: number;
  insumo_id: number;
  quantidade_realizada: number;
  unidade_medida: string;
  observacoes: string | null;
  justificativa_excesso: string | null;
  created_at: string;
}

export interface ConsumoInsumoMissaoCreate {
  missao_id: number;
  insumo_id: number;
  quantidade_realizada: number;
  unidade_medida: string;
  observacoes?: string | null;
  justificativa_excesso?: string | null;
}

File: app/schemas/reserva_insumo.py
"""Pydantic schemas for ReservaInsumo and ConsumoInsumoMissao."""

from datetime import datetime
from decimal import Decimal

from pydantic import BaseModel, Field


class ReservaInsumoBase(BaseModel):
    missao_id: int
    insumo_id: int
    quantidade_prevista: Decimal = Field(..., ge=0)
    unidade_medida: str = Field(..., max_length=30)
    justificativa_excesso: str | None = None


class ReservaInsumoCreate(ReservaInsumoBase):
    pass


class ReservaInsumoResponse(ReservaInsumoBase):
    id: int
    created_at: datetime

    model_config = {"from_attributes": True}


class ConsumoInsumoMissaoBase(BaseModel):
    missao_id: int
    insumo_id: int
    quantidade_realizada: Decimal = Field(..., ge=0)
    unidade_medida: str = Field(..., max_length=30)
    observacoes: str | None = None
    justificativa_excesso: str | None = None


class ConsumoInsumoMissaoCreate(ConsumoInsumoMissaoBase):
    pass


class ConsumoInsumoMissaoResponse(ConsumoInsumoMissaoBase):
    id: int
    created_at: datetime

    model_config = {"from_attributes": True}
